sqrl-planner/sqrl-planner-local/src/main/java/com/datasqrl/graphql/generate/ObjectTypeGenerator.java

```diff
@@ -40,20 +43,33 @@ public Set<GraphQLType> visit(SqrlSchemaForInference schema, SchemaGeneratorCont
 
   @Override
   public GraphQLType visit(SQRLTable table, SchemaGeneratorContext context) {
+    List<GraphQLFieldDefinition> fields = table.getFields(true)
+        .stream()
+        .filter(f -> !(f instanceof Column) || getOutputType(((Column) f).getType()).isPresent())
+        .map(f -> f.accept(this, context))
+        .collect(Collectors.toList());
+
+    //Include hidden fields in relation has none.
+    if (fields.isEmpty()) {
+      fields = table.getFields(false)
+          .stream()
+          .filter(f->!f.getName().getDisplay().startsWith(SYSTEM_HIDDEN_PREFIX))
+          .filter(f -> !(f instanceof Column) || getOutputType(((Column) f).getType()).isPresent())
+          .map(f -> f.accept(this, context))
+          .collect(Collectors.toList());
+    }
+
     return GraphQLObjectType.newObject()
         .name(getTypeName(table, context.getNames()))
-        .fields(table.getFields(true)
-            .stream()
-            .map(f -> f.accept(this, context))
-            .collect(Collectors.toList()))
+        .fields(fields)
         .build();
   }
 
   @Override
   public GraphQLFieldDefinition visit(Column column, SchemaGeneratorContext context) {
     return GraphQLFieldDefinition.newFieldDefinition()
-        .name(column.getName().getDisplay())
-        .type(wrap(getOutputType(column.getType()), column.getType()))
+        .name(conformName(column.getName().getDisplay()))
```

Comment:
This is a source of bugs. We'll need to either canonicalize the names or prevent them from being defined.

Reply:
Aren't column names already canonicalized by the NameCanonicalizer?